app/assets/javascripts/controllers/checkout_controller.coffee

```diff
@@ -79,6 +80,22 @@ app.controller 'CheckoutController',['$window', '$q', '$scope', 'Order', 'Addres
       .finally ->
         $scope.savingAddress = false
 
+   $scope.applyCoupon = (couponCode) ->
+    return if $scope.grabbingCoupon
+    $scope.grabbingCoupon = true
+    Coupon.query(code: couponCode).$promise
+      .then (data) ->
+        coupon = new Coupon(data)
+        itemToSave = new OrderItem(quantity: 1, source_type: "Coupon", source_id: coupon.id, order_id: $scope.cartOrder.id )
+        itemToSave.$save().then ->
```

Comment:
do you want to do `itemToSave.$save().then (item) ->`? This way you get the updated item with its id and what not.

Reply:
stop making so much sense